tests/integration/test_cubepart.py

```diff
@@ -902,6 +902,26 @@ def it_can_sort_by_column_percent(self):
         actual = np.round(slice_.column_percentages, 1).tolist()
         assert expected == actual, "\n%s\n\n%s" % (expected, actual)
 
+    def it_can_sort_by_value_for_mr_derived_insertions(self):
+        transforms = {
+            "rows_dimension": {
+                "order": {
+                    "direction": "ascending",
+                    "type": "opposing_insertion",
+                    "insertion_id": 1,
+                    "measure": "count_weighted",
+                }
+            }
+        }
+        slice_ = Cube(MRI.CAT_X_MR, transforms=transforms).partitions[0]
+        assert slice_.counts.tolist() == [
+            [0.0, 0.0, 0.0, 0.0],
+            [0.0, 0.0, 0.0, 0.0],
+            [0.6, 0.0, 0.6, 0.0],
+            [0.9, 0.9, 0.0, 0.0],
+            [1.5, 1.5, 1.5, 0.0],
+        ]
```

Comment:
It's tricky to see what is what here. Could you please annotate (like with comments) what is what, and what we should expect? Also, I always find it very useful to have an expectation for the case with "no transforms" alongside the one "with transforms", preferably in the same test, which makes it much easier to compare against.

Reply:
Ha, I really like comparisons too, but in an earlier PR (https://github.com/Crunch-io/crunch-cube/pull/300#discussion_r632144756) Steve said he didn't like those comparisons. Do you have advice on making it more appetizing for Steve? Otherwise, are comments the happy medium?

Reply:
I'll bring it up in standup, or at the meeting afterwards. Maybe a comment after each line, representing the original order, or something. I really feel I need this as a reader of the tests.

Reply:
I don't have any problem at all with adding the untransformed order as a comment like:

```python
# --- payload order is ---
# [1.5, 1.5, 1.5, 0.0],
# [0.6, 0.0, 0.6, 0.0],
# [0.0, 0.0, 0.0, 0.0],
# [0.9, 0.9, 0.0, 0.0],
# [0.0, 0.0, 0.0, 0.0],
```

It's just that it muddles the test for me if you're actually _testing_ that the untransformed order is what you expect. Either that needs to be tested or it doesn't (perhaps because it's already tested well enough in other tests, which we would expect to be 95%+ the case). If it _doesn't_ need to be tested, then you have redundancy in the tests which consumes test runtime and increases the cognitive load of the tests.

More importantly for me, you need two fixturings, which breaks the one fixture per test pattern which makes tests quick to understand (because they look like all the other tests) and compact (because there's only one fixture and one expectation.

Also, you can hide a test failure. If the first (redundant) test fails, it prevents the second (real) test from running. If they are separate, the "real" test will be run in any case.

I'm sure there are other reasons, but in any case, I don't have a problem using comments containing baseline (untransformed in this case) results to clarify what's happening. I just don't like the idea of using "actual" tests just for this documentation purpose.

Reply:
One other argument, just while I'm thinking about it. Say you had a scenario where col-percents were not appearing properly when in payload order, but were showing up fine when sorted by a derived subvariable or whatever. Then that test would fail saying: `it_sorts_column_percents_in_derived_subvar_order FAILED` whereas the actual problem is more like `it_displays_column_percents_in_payload_order FAILED`. So the name and whatever other documentation focused on the impetus of the test is potentially made misleading because you tested something other than what you said you were testing (before testing the real thing).

Reply:
Added a comment indicting which column is the MR insertion. I don't like having the whole cube as a comment because I worry about comments going stale. Hope that's a good enough compromise.

I'll follow the established conventions here, but I'm not super convinced by these arguments. My main argument against comments is that I worry about them going stale. And personally, I don't see them as cognitive overhead. To me, these intermediate tests are more like breadcrumbs that help someone in the future more quickly understand what is being tested. And problems with "hidden" or "misnamed" tests are just inherent to integration tests (what if the JSON parser broke?), so don't seem that problematic to me.

If you know of further reading on this, I'd be interested to read!

Reply:
Yeah, I suppose I'm okay with having before and after. I don't like them, and I don't think I would write one like that myself, but if you feel like they make the test easier to make sense of then I won't stand in the way.

I find the point about guaranteeing the "before" version hasn't gotten out of date pretty compelling.

I would like this example to have more different numbers though, you can't tell whether the sort is stable because you can't distinguish between the top rows that are all 0s. It's good to have two same sort values so you demonstrate what happens when values are equal, but the rest of the row needs to be distinguishable from the other row with the same sort value.